LoopKit/TemporaryScheduleOverrideSettings.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  TemporaryScheduleOverrideSettings.swift
+//  LoopKit
+//
+//  Created by Michael Pangburn on 1/2/19.
+//  Copyright © 2019 LoopKit Authors. All rights reserved.
+//
+
+import Foundation
+
+
+public struct TemporaryScheduleOverrideSettings: Hashable {
+    public var targetRange: DoubleRange?
+    public var insulinNeedsScaleFactor: Double?
+
+    public var basalRateMultiplier: Double? {
+        return insulinNeedsScaleFactor
+    }
+
+    public var insulinSensitivityMultiplier: Double? {
+        return insulinNeedsScaleFactor.map { 1.0 / $0 }
+    }
+
+    public var carbRatioMultiplier: Double? {
+        return insulinNeedsScaleFactor.map { 1.0 / $0 }
+    }
+
+    public var effectiveInsulinNeedsScaleFactor: Double {
+        return insulinNeedsScaleFactor ?? 1.0
+    }
+
+    public init(targetRange: DoubleRange?, insulinNeedsScaleFactor: Double? = nil) {
+        self.targetRange = targetRange
+        self.insulinNeedsScaleFactor = insulinNeedsScaleFactor
+    }
+}
+
+extension TemporaryScheduleOverrideSettings: RawRepresentable {
+    public typealias RawValue = [String: Any]
+
+    private enum Key {
+        static let targetRange = "targetRange"
```

Comment:
I don't have a strong preference. What we probably *should* do is move away from writing these serializations ourselves and instead start adopting Codable to do the work for us.

Reply:
Yes, that would make sense. Do you think it would have to be an all or none type adoption?

Reply:
It wouldn't have to be all-or-none. Getting free Codable synthesis just requires that all stored properties are themselves Codable; this will probably be more convenient for some types than for others. I don't think migrating all of our manual serializations would be worth the headache, but new data types might as well take advantage of the feature where they can.